test/unit/controllers/webhook.test.js

```diff
@@ -0,0 +1,751 @@
+    mockGetReviewFn.mockImplementation((mergeReqNbr) => new Promise((resolve, reject) => resolve(mockReview)))
+
+    const errorMsg = 'process merge error msg'
+    mockProcessMergeFn.mockImplementation((mergeReqNbr) => new Promise((resolve, reject) => reject(new Error(errorMsg))))
+
+    expect.hasAssertions()
+    return webhook(req, res).then(response => {
+      if (service === 'github') {
+        expect(mockCreateHmacFn).toHaveBeenCalledTimes(1)
+      }
+      expect(mockCreateFn).toHaveBeenCalledTimes(1)
+      expect(mockGetReviewFn).toHaveBeenCalledTimes(1)
+      expect(mockProcessMergeFn).toHaveBeenCalledTimes(1)
+      // Shold still attempt to delete branch despite error sending notification emails.
+      if (service === 'github') {
+        expect(mockDeleteBranchFn).toHaveBeenCalledTimes(1)
+      } else if (service === 'gitlab') {
+        expect(mockDeleteBranchFn).toHaveBeenCalledTimes(0)
+      }
+      expect(res.send.mock.calls[0][0]).toEqual({ errors: '[\"' + errorMsg + '\"]' })
+      expect(res.status.mock.calls[0][0]).toBe(400)
+    })
+  })
+
+  test.each([
+    ['github']
+  ])('abort and return an error if delete branch fails - %s', async (service) => {
+    req.params.service = service
+
+    req.body = {
+      number: 123,
+      object_attributes: {
+        iid: 234
+      }
+    }
+
+    req.headers['x-github-event'] = 'pull_request'
+    mockGetSiteConfigFn.mockImplementation(() => new Promise((resolve, reject) => resolve(new Map([
+        ['githubWebhookSecret', 'sha1=' + mockHmacDigest]
+      ])))
+    )
+    req.headers['x-hub-signature'] = 'sha1=' + mockHmacDigest
+
+    mockReview = new Review('Some random PR', sampleData.prBody1, 'merged', 'staticman_1234567', 'master')
+    mockGetReviewFn.mockImplementation((mergeReqNbr) => new Promise((resolve, reject) => resolve(mockReview)))
+    mockProcessMergeFn.mockImplementation((mergeReqNbr) => new Promise((resolve, reject) => resolve(true)))
+
+    const errorMsg = 'delete branch error msg'
+    mockDeleteBranchFn.mockImplementation((sourceBranch) => new Promise((resolve, reject) => reject(errorMsg)))
+
+    // Suppress any calls to console.error - to keep test output clean.
+    const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    expect.hasAssertions()
+    return webhook(req, res).then(response => {
+      expect(mockCreateHmacFn).toHaveBeenCalledTimes(1)
+      expect(mockCreateFn).toHaveBeenCalledTimes(1)
+      expect(mockGetReviewFn).toHaveBeenCalledTimes(1)
+      expect(mockProcessMergeFn).toHaveBeenCalledTimes(1)
+      expect(mockDeleteBranchFn).toHaveBeenCalledTimes(1)
+      expect(res.send.mock.calls[0][0]).toEqual(
+        { errors: '[\"Failed to delete merge branch ' + mockReview.sourceBranch + ' - ' + errorMsg + '\"]' })
+      expect(res.status.mock.calls[0][0]).toBe(400)
+
+      // Restore console.error
+      consoleSpy.mockRestore();
```

Comment:
This should probably be done in a `.finally()` just in case an exception is thrown. Same applies to other mock restores done in `.then()` blocks